New plugin creater

SimonBaeumer opened the pull request:
Supporting the `5.2` and `legacy` plugin system for `sw plugin:create`.
Working on top of @mohrekopp pull request: #70. 

Added:
- Move `Views/` to `Resources/views`
- Move `Snippets/` to `Resources/snippets`
- `Resources/menu.xml`
- `Resources/config.xml`
- Service registrations in `Resources/services.xml`
- Create widget
- Separate directories for `legacy` and `5.2` plugin system
- Little unit test fixes

Thanks @mohrekopp for your Pull-Request #70, these big changes must be done due to missing documentation at the moment.

FiveDigital commented:
Shouldn't Commands and Subscribers be written in the `services.xml` with their according tags instead of adding methods in the plugin class for them?

mitelg commented on src/Extensions/Shopware/PluginCreator/template/current/PluginClass.tpl:
```diff
@@ -0,0 +1,138 @@
+     *
+     * @param Application $application
+     */
+    public function registerCommands(Application $application)
```
is this necessary?

mohrekopp replied:
IMHO we should actually remove this part

SimonBaeumer replied:
Agreed 100% ;)

mitelg commented:
@FiveDigital commands and subscribers are registered in the services.xml if I see this correctly. the onStartDispatch event in the plugin.php is only used for registering the API components

FiveDigital commented:
The according part was removed exactly one minute after my comment with https://github.com/shopwareLabs/sw-cli-tools/pull/71/files/1f5edd94f005e0ff43386f9bd8ea204fe851216b..9b4004c391cafc68622126a0ccb114d7dc4da4e0 👍

SimonBaeumer commented:
I improved the pull-request a little bit more by adding the `Subscriber/ApiSubscriber.php` and better support for the command arguments.

OliverSkroblin commented:
1. new Plugins created in root directory

2. install and uninstall function in plugin class are empty, unnecessary boiler plate code (php storm override function generates this code within 2sec )

3. controller path service definition exists even if when the controllerPath subscriber class not exist.

4. --have-filter 
	4.0 please consider SearchBundleES too
	4.1 Add a decorator for ESIndexingBundle/Product/*
	4.2 Add a sorting too please, with template extension  
	4.3 move CriteriaRequestHandler into SearchBundle namespace
5. --have-widget
	5.0 Mixed with backend module implementation
	5.2 ControllerPath Subscriber contains template extends command for backend widgets => extract
	5.3 Backend/Application controller contains php code for backend widget
	5.4 views/backend/application/widgets contains extjs widgets 
	==> please split in seperate namespace
6. $this->container->get('application')->Loader() => $this->container->get('Loader');

SimonBaeumer commented:
@OliverSkroblin The `1.` and `2.` point of your list should be done in additional PRs, because this PR just handles  the compatibility to the `5.2` plugin system. 
In this case the target installation directory and SearchBundleES integration would be new functionality.

All other points on your list were fixed.

SimonBaeumer commented:
Added:
 
 - Root folder detection, created plugins will be created in the plugin path
  - legacy plugins in `engine/Shopware/Plugins/Local/your_namespace' 
  - `5.2` plugins in `custom/plugins`
  - if the command will not be executed from shopware root it will be created in the same directory
- Added elastic search integration
- Added product search decorator
- Added sorting

SimonBaeumer commented:
This PR is ready to merge. 
Thank you for your contribution, @mohrekopp ;)
